opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -141,18 +146,41 @@ namespace Opm
                 if (numSamples > 0) {
                     auto splinePvt = std::shared_ptr<PvtDeadSpline>(new PvtDeadSpline);
                     splinePvt->initFromOil(pvdoTables, numSamples);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        splinePvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                     deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = splinePvt;
                 } else {
                     auto deadPvt = std::shared_ptr<PvtDead>(new PvtDead);
                     deadPvt->initFromOil(pvdoTables);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        deadPvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                   deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = deadPvt;
                 }
             } else if (!pvtoTables.empty()) {
-                props_[phase_usage_.phase_pos[Liquid]].reset(new PvtLiveOil(pvtoTables));
```

Comment:
Fine, I did not notice that they also were different. If you'd like to make that one too like this, feel free, or tell me if you are done with this. It is ready to be merged anyway.

The reason I prefer the shared rather than plain pointer is that it eliminates future bugs if someone were to introduce some possibly throwing operation between what is now consecutive lines:
```
pvto->modificationThatMayThrow();
```
It also gives that fuzzy good feeling of having avoided plain-but-owning pointers...